## About Dataset

##### One way to understand how a city government works is by looking at who it employs and how its employees are compensated. This data contains the names, job title, and compensation for San Francisco city employees on an annual basis from 2011 to 2014.


In [1]:
import  pandas as pd
print("Importing Panda Completed")

Importing Panda Completed


In [2]:
data = pd.read_csv('Sf Salary Dataset.csv')

C:\Users\adv\AppData\Local\Temp\ipykernel_3220\439286055.py:1: DtypeWarning: Columns (3,4,5,6,12) have mixed types. Specify dtype option on import or set low_memory=False.
  data = pd.read_csv('Sf Salary Dataset.csv')


#### Finding Number of Rows And Column (Shape OF dataset)


In [6]:
rows = data.shape[0]
col = data.shape[1]
print("No. of Rows : "+str(rows) +"\nNo. of Column : "+str(col))

No. of Rows : 148654
No. of Column : 13


#### Info of dataset like Column data type


In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 148654 entries, 0 to 148653
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Id                148654 non-null  int64  
 1   EmployeeName      148654 non-null  object 
 2   JobTitle          148654 non-null  object 
 3   BasePay           148049 non-null  object 
 4   OvertimePay       148654 non-null  object 
 5   OtherPay          148654 non-null  object 
 6   Benefits          112495 non-null  object 
 7   TotalPay          148654 non-null  float64
 8   TotalPayBenefits  148654 non-null  float64
 9   Year              148654 non-null  int64  
 10  Notes             0 non-null       float64
 11  Agency            148654 non-null  object 
 12  Status            38119 non-null   object 
dtypes: float64(3), int64(2), object(8)
memory usage: 14.7+ MB


#### Checking Null Values


In [ ]:

data.isnull().sum()


Id                       0
EmployeeName             0
JobTitle                 0
BasePay                605
OvertimePay              0
OtherPay                 0
Benefits             36159
TotalPay                 0
TotalPayBenefits         0
Year                     0
Notes               148654
Agency                   0
Status              110535
dtype: int64

#### Here Its shows That No. of Null values in Each Column Of Dataset

#### i.e. 605 Null values in BasePay Column and so on


#### Droping Notes,Agency and Status Column because it not requred Further


In [11]:
data = data.drop(['Notes','Agency','Status'],axis=1)

In [12]:
data = data.drop(['Id'],axis=1)

#### Overall Statics Of dataset


In [16]:
data.describe(include='all')

EmployeeName          JobTitle   BasePay  OvertimePay  OtherPay  \
count        148654            148654  148049.0     148654.0  148654.0   
unique       110811              2159  109900.0      66555.0   84968.0   
top       Kevin Lee  Transit Operator       0.0          0.0       0.0   
freq             13              7036     875.0      66103.0   35218.0   
mean            NaN               NaN       NaN          NaN       NaN   
std             NaN               NaN       NaN          NaN       NaN   
min             NaN               NaN       NaN          NaN       NaN   
25%             NaN               NaN       NaN          NaN       NaN   
50%             NaN               NaN       NaN          NaN       NaN   
75%             NaN               NaN       NaN          NaN       NaN   
max             NaN               NaN       NaN          NaN       NaN   

        Benefits       TotalPay  TotalPayBenefits           Year  
count   112495.0  148654.000000     148654.000000  148654.000000  
unique   99635.0            NaN               NaN            NaN  
top          0.0            NaN               NaN            NaN  
freq      1053.0            NaN               NaN            NaN  
mean         NaN   74768.321972      93692.554811    2012.522643  
std          NaN   50517.005274      62793.533483       1.117538  
min          NaN    -618.130000       -618.130000    2011.000000  
25%          NaN   36168.995000      44065.650000    2012.000000  
50%          NaN   71426.610000      92404.090000    2013.000000  
75%          NaN  105839.135000     132876.450000    2014.000000  
max          NaN  567595.430000     567595.430000    2014.000000

#### Find The Top 5 Employee Name Occurance


In [17]:
top5En=data['EmployeeName'].value_counts().head(5)
top5En

EmployeeName
Kevin Lee       13
Richard Lee     11
Steven Lee      11
William Wong    11
Stanley Lee      9
Name: count, dtype: int64

#### No. Unique job title


In [20]:
print("No. of Unique Job Column:",data['JobTitle'].nunique())


No. of Unique Job Column: 2159


#### Total Number of Job titles Contain captain


In [27]:
jobCaptain=len(data[data['JobTitle'].str.contains("Captain",case=False)])
print("Job Title Having Captain:",jobCaptain)

Job Title Having Captain: 552


#### Names Of All Employee From Fire department


In [30]:
data[data['JobTitle'].str.contains("Fire",case=False)]['EmployeeName']

4            PATRICK GARDNER
6                  ALSON LEE
8             MICHAEL MORRIS
9         JOANNE HAYES-WHITE
10             ARTHUR KENNEY
                 ...        
145956      Kenneth C Farris
147556         Edward A Dunn
148021        Kari A Johnson
148209          Sheryl K Lee
148554       Lawrence F Gatt
Name: EmployeeName, Length: 5879, dtype: object

#### Finding Minimum,maximum & Avergae base Pay


In [47]:
data['BasePay'] = pd.to_numeric(data['BasePay'],errors='coerce')
maxi = float(data['BasePay'].max())
mini = float(data['BasePay'].min())
avg = round(float(data['BasePay'].mean()),2)
print("The Maximum Basepay is :",maxi)
print("The Minimum Basepay is :",mini)
print("The Average Basepay is :",avg)

The Maximum Basepay is : 319275.01
The Minimum Basepay is : -166.01
The Average Basepay is : 66325.45


#### Replacing Not Provided NaN in Employee Name


In [7]:
import numpy as np
data['EmployeeName']=data['EmployeeName'].replace('Not provided',np.nan)
data['EmployeeName']

0            NATHANIEL FORD
1              GARY JIMENEZ
2            ALBERT PARDINI
3         CHRISTOPHER CHONG
4           PATRICK GARDNER
                ...        
148649        Roy I Tillery
148650                  NaN
148651                  NaN
148652                  NaN
148653            Joe Lopez
Name: EmployeeName, Length: 148654, dtype: object

In [26]:
import numpy as np
data['BasePay']=data['BasePay'].replace('Not Provided',np.nan)
data

EmployeeName                                        JobTitle  \
0          NATHANIEL FORD  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY   
1            GARY JIMENEZ                 CAPTAIN III (POLICE DEPARTMENT)   
2          ALBERT PARDINI                 CAPTAIN III (POLICE DEPARTMENT)   
3       CHRISTOPHER CHONG            WIRE ROPE CABLE MAINTENANCE MECHANIC   
4         PATRICK GARDNER    DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)   
...                   ...                                             ...   
148649      Roy I Tillery                                       Custodian   
148650                NaN                                    Not provided   
148651                NaN                                    Not provided   
148652                NaN                                    Not provided   
148653          Joe Lopez                      Counselor, Log Cabin Ranch   

          BasePay OvertimePay   OtherPay Benefits   TotalPay  \
0       167411.18         0.0  400184.25      NaN  567595.43   
1       155966.02   245131.88  137811.38      NaN  538909.28   
2       212739.13   106088.18    16452.6      NaN  335279.91   
3         77916.0    56120.71   198306.9      NaN  332343.61   
4        134401.6      9737.0  182234.59      NaN  326373.19   
...           ...         ...        ...      ...        ...   
148649       0.00        0.00       0.00     0.00       0.00   
148650        NaN         NaN        NaN      NaN       0.00   
148651        NaN         NaN        NaN      NaN       0.00   
148652        NaN         NaN        NaN      NaN       0.00   
148653       0.00        0.00    -618.13     0.00    -618.13   

        TotalPayBenefits  Year  
0              567595.43  2011  
1              538909.28  2011  
2              335279.91  2011  
3              332343.61  2011  
4              326373.19  2011  
...                  ...   ...  
148649              0.00  2014  
148650              0.00  2014  
148651              0.00  2014  
148652              0.00  2014  
148653           -618.13  2014  

[148654 rows x 9 columns]

#### Droping the Rows Having 5 Missing Values


In [33]:
data.drop(data[data.isnull().sum(axis=1) == 5].index,axis=0,inplace=True)
data


EmployeeName                                        JobTitle  \
0          NATHANIEL FORD  GENERAL MANAGER-METROPOLITAN TRANSIT AUTHORITY   
1            GARY JIMENEZ                 CAPTAIN III (POLICE DEPARTMENT)   
2          ALBERT PARDINI                 CAPTAIN III (POLICE DEPARTMENT)   
3       CHRISTOPHER CHONG            WIRE ROPE CABLE MAINTENANCE MECHANIC   
4         PATRICK GARDNER    DEPUTY CHIEF OF DEPARTMENT,(FIRE DEPARTMENT)   
...                   ...                                             ...   
148645   Carolyn A Wilson                       Human Services Technician   
148647     Joann Anderson                     Communications Dispatcher 2   
148648        Leon Walker                                       Custodian   
148649      Roy I Tillery                                       Custodian   
148653          Joe Lopez                      Counselor, Log Cabin Ranch   

          BasePay OvertimePay   OtherPay Benefits   TotalPay  \
0       167411.18         0.0  400184.25      NaN  567595.43   
1       155966.02   245131.88  137811.38      NaN  538909.28   
2       212739.13   106088.18    16452.6      NaN  335279.91   
3         77916.0    56120.71   198306.9      NaN  332343.61   
4        134401.6      9737.0  182234.59      NaN  326373.19   
...           ...         ...        ...      ...        ...   
148645       0.00        0.00       0.00     0.00       0.00   
148647       0.00        0.00       0.00     0.00       0.00   
148648       0.00        0.00       0.00     0.00       0.00   
148649       0.00        0.00       0.00     0.00       0.00   
148653       0.00        0.00    -618.13     0.00    -618.13   

        TotalPayBenefits  Year  
0              567595.43  2011  
1              538909.28  2011  
2              335279.91  2011  
3              332343.61  2011  
4              326373.19  2011  
...                  ...   ...  
148645              0.00  2014  
148647              0.00  2014  
148648              0.00  2014  
148649              0.00  2014  
148653           -618.13  2014  

[148650 rows x 9 columns]

#### Job Title of Albert Pardani


In [34]:
data[data['EmployeeName'].str.contains('ALBERT PARDINI')]['JobTitle']

2    CAPTAIN III (POLICE DEPARTMENT)
Name: JobTitle, dtype: object

#### Albert PArdani Make including Benefits


In [38]:
amount = data[data['EmployeeName']=='ALBERT PARDINI']['TotalPayBenefits'].values[0]
print("The Toala Money Albert pandani make is :",amount)

The Toala Money Albert pandani make is : 335279.91


#### Display Name of Person Having Highest base Pay


In [43]:
high = data[data['BasePay'].max()==data['BasePay']]['EmployeeName'].values[0]
high

'Gregory P Suhr'

#### Find Average BasePay of All Employee Per Year


In [51]:

data.groupby('Year')['BasePay'].mean()

Year
2011    63595.956517
2012    65436.406857
2013    69630.030216
2014    66564.421924
Name: BasePay, dtype: float64

#### Average basepay as per Job Title


In [52]:
data.groupby('JobTitle')['BasePay'].mean()

JobTitle
ACCOUNT CLERK                                     43300.806506
ACCOUNTANT                                        46643.172000
ACCOUNTANT INTERN                                 28732.663958
ACPO,JuvP, Juv Prob (SFERS)                       62290.780000
ACUPUNCTURIST                                     66374.400000
                                                      ...     
X-RAY LABORATORY AIDE                             47664.773077
X-Ray Laboratory Aide                             46086.387100
YOUTH COMMISSION ADVISOR, BOARD OF SUPERVISORS    52609.910000
Youth Comm Advisor                                39077.957500
ZOO CURATOR                                       43148.000000
Name: BasePay, Length: 2158, dtype: float64

#### Average Base Pay For job Title ACCOUNTANT


In [58]:
basepay = data[data['JobTitle'] == "ACCOUNTANT"]['BasePay'].mean()
float(basepay)

46643.172

#### Top Common Jobs


In [59]:
data['JobTitle'].value_counts().head(5)

JobTitle
Transit Operator                7036
Special Nurse                   4389
Registered Nurse                3736
Public Svc Aide-Public Works    2518
Police Officer 3                2421
Name: count, dtype: int64

#### Here All The Question Completed
